web-app/js/portal/cart/DownloadPanelItemTemplate.js

```diff
@@ -86,10 +86,33 @@ Portal.cart.DownloadPanelItemTemplate = Ext.extend(Ext.XTemplate, {
     },
 
     _downloadButton: function(collection) {
-        this._createDownloadButton.defer(1, this, [collection]);
+
+        if (collection.downloadStatus == 'requested') {
+            this._createDownloadingLabel.defer(1, this, [collection]);
+        }
+        else {
+            this._createDownloadButton.defer(1, this, [collection]);
+        }
+
         return '';
     },
 
+    // It's *actually* a button, but we're using it as a label here...
+    _createDownloadingLabel: function(collection) {
+        var elementId = this._getButtonId(collection, 'downloadButtonId');
+
+        Ext.fly(elementId).update("");
+
+        new Ext.Button({
+            text: OpenLayers.i18n('downloadStatusRequested'),
+            icon: 'images/spinner_orange_background.gif',
```

Comment:
@jkburges 
I gave you a bum steer on the spinner. It turns out that the ' fa-spin' class must be added. 
The disabled class can be overridden. Below works for me


```
        new Ext.Button({
            text: "<span class=\"fa fa-spin fa-spinner \"></span>" + OpenLayers.i18n('downloadStatusRequested'),
            cls: 'navigationButton navigationButtonActive',
            scope: this,
            disabled: true,
            disabledClass: '',
            renderTo: elementId
        });
```
CSS
```
.navigationButtonActive.x-btn {
    background-color: #89C8EE;
}
```